contrib/platform/test/com/sun/jna/platform/win32/Kernel32Test.java

```diff
@@ -383,6 +381,51 @@ public void testQueryFullProcessImageName() {
         }
     }
 
+    public void testGetProcessTimes() {
+        int pid = Kernel32.INSTANCE.GetCurrentProcessId();
+        HANDLE h = Kernel32.INSTANCE.OpenProcess(WinNT.PROCESS_QUERY_INFORMATION, false, pid);
+        assertNotNull("Failed (" + Kernel32.INSTANCE.GetLastError() + ") to get process ID=" + pid + " handle", h);
+
+        try {
+            FILETIME lpCreationTime = new FILETIME();
+            FILETIME lpExitTime = new FILETIME();
+            FILETIME lpKernelTime = new FILETIME();
+            FILETIME lpUserTime = new FILETIME();
+            boolean b = Kernel32.INSTANCE.GetProcessTimes(h, lpCreationTime, lpExitTime, lpKernelTime, lpUserTime);
+            assertTrue("Failed (" + Kernel32.INSTANCE.GetLastError() + ") to get process times", b);
+            // Process must have started after 1970 and before now.
+            assertTrue(lpCreationTime.toTime() > 0);
+            assertTrue(lpCreationTime.toTime() <= System.currentTimeMillis());
+            // lpExitTime is undefined for a running process, do not test
+            // Kernel and User time must be >= 0
+            assertTrue(lpKernelTime.toTime() >= 0);
+            assertTrue(lpUserTime.toTime() >= 0);
```

Comment:
The asserts in lines 401 and 402 fail for me. The conversion in `WinBase#FILETIME` is implemented under the assumption, that a `FILETIME ` is a timestamp. In this case this is wrong. the kernel and user times are relative times, that the process accumulated in user-/kernelmode. Two options: use `FILEMTIME#toDWordLong` or add a conversion to `long` to `FILEMTIME`.

The method binding for `GetProcessTimes` looks good, it is just the test, that das not work.